java/arcs/sdk/android/storage/ServiceStore.kt

```diff
@@ -159,15 +169,27 @@ class ServiceStore<Data : CrdtData, Op : CrdtOperation, ConsumerData>(
     override fun off(callbackToken: Int) {
         val service = checkNotNull(storageService)
         runBlocking {
-            channel.send { service.unregisterCallback(callbackToken) }
+            send {
+                service.unregisterCallback(callbackToken)
+                initChannel()
+            }
+        }
+    }
+
+    private suspend fun send(block: suspend () -> Unit) = requireNotNull(channel) {
+        "Channel is not initialized"
+    }.apply {
+        require(!isClosedForSend) {
+            "Channel is closed"
```

Comment:
Something in this change is causing flakyness see eg
https://g3c.corp.google.com/results/invocations/5f5d3df9-ce0d-4788-a92f-2be6f167fe56/targets/%2F%2Fthird_party%2Fjava_src%2Farcs%2Fjavatests%2Farcs%2Fandroid%2Fentity:DifferentHandleManagerTest/tests